react/hocs/WithBlocksGridCount/index.js

```diff
@@ -0,0 +1,28 @@
+import React, { Component } from 'react';
+import { SizeMe } from 'react-sizeme';
+
+import constants from 'react/styles/constants';
+
+const withBlocksGridCount = (WrappedComponent) => {
+  class WithBlocksGridCount extends Component {
+    render() {
+      const { ...rest } = this.props;
+
+      return (
+        <SizeMe>
+          {({ size }) => {
+            const divider = parseInt(constants.blockAndGutter, 10);
+            const blocksGridCount = Math.floor(size.width / divider);
+            return (
+              <WrappedComponent blocksGridCount={blocksGridCount} {...rest} />
+            );
+          }}
+        </SizeMe>
```

Comment:
Why is any of this necessary?

Reply:
The block count on the bottom right is dependent on how many blocks are visible on the screen.

Reply:
This seems like a super over kill way of going about it. Couldn't you just check the el offsets?

Reply:
What does this do on server vs client-side render?

Reply:
Because presumably no matter what it's going to be wrong on initial render. Shouldn't we just design around this?

Reply:
Alright agreed. Will confer with Chris